src/services/interfaces/IBaseServiceInterface.java

```diff
@@ -0,0 +1,16 @@
+package services.interfaces;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public interface IBaseServiceInterface<T, E> {
+  Optional<T> insert(T object);
```

Comment:
I don't get T and E. I think stands for entity but here seems to be the key?

Reply:
solved